### DISCUSSION TITLE
Lots of Busy Try again later red popups when tagging

### DISCUSSION THREAD

**Discussion by RobPattonDotCom** [photoprism/photoprism#3158] [Q&A]
Docker  Host is 10 cores and 64 gig ram
Using Docker MariaDB for database

85K pics scanned, 4 days later I'm tagging faces

About 70% of the time adding a name either won't let me type anything, or, if I can, I get a Busy try again later.  If I just sit for 60 secs, it eventually seems to wake up and let me save the name, but EVERY time I gotta go back and check to make sure it saves, which is annoying, and makes the process take a long long time.  Sometimes it works perfectly, and its great, but thats the exception

None of my monitoring indicates that the containers are even remotely overloaded, even so, I've ramped up the resources with no notable change.

Logs don't tell me anything that helps (ME) but I do see a lot of this

time="2023-01-31T18:43:58Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:44:05Z" level=info msg="faces: resolved 2 conflicts"
time="2023-01-31T18:44:06Z" level=info msg="changes successfully saved"
time="2023-01-31T18:44:07Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:44:15Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:44:22Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:44:29Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:44:37Z" level=info msg="faces: merged 55 clusters [1m22.836135636s]"
time="2023-01-31T18:44:48Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:45:11Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:45:34Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:45:49Z" level=info msg="faces: merged 53 clusters"
time="2023-01-31T18:45:49Z" level=info msg="changes successfully saved"
time="2023-01-31T18:49:40Z" level=info msg="subject: added person 'Howard Scott Warshaw'"
time="2023-01-31T18:49:56Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:50:19Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:50:42Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:51:05Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:51:14Z" level=info msg="faces: merged 51 clusters"
time="2023-01-31T18:51:14Z" level=info msg="changes successfully saved"
time="2023-01-31T18:51:36Z" level=info msg="metadata: updated 28 photos"
time="2023-01-31T18:52:12Z" level=info msg="subject: added person 'Billie Taylor'"
time="2023-01-31T18:52:29Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:52:52Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:53:15Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:53:39Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:53:46Z" level=info msg="faces: merged 51 clusters"
time="2023-01-31T18:53:46Z" level=info msg="changes successfully saved"
time="2023-01-31T18:58:45Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:59:10Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:59:30Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:59:53Z" level=warning msg="faces: failed removing merged clusters for subject jrp5ll02unhrzjcj"
time="2023-01-31T18:59:53Z" level=info msg="faces: merged 51 clusters [1m22.831019732s]"

**Comment by RobPattonDotCom** [comment]
To add more info.  Scrolling through pics is SUPER FAST.  It seems to only be the name tagging part that is slow.

**Reply by lastzero** [reply]
Note we already have an issue for possibly the very same issue. Once I have a bit of time, I will look into it.

**Reply by lastzero** [reply]
Did you start assigning faces to people before indexing was complete?

**Reply by RobPattonDotCom** [reply]
Yes, I think so.   I rand this process once before, and after 4 days had this same problem, so I tore down the containers, and deployed with an external db server thinking I was just overloading the builtin db.    The first attempt I did a lot. This 2nd try, I tagged only myself, and didnt try to tag everyone.

**Reply by lastzero** [reply]
So the problem could be a conflict of the manual assignments during indexing and the automated clustering that happens once indexing is complete. We always wait when testing, so that would explain why we never observed anything like that.

**Reply by lastzero** [reply]
Note you don't need to delete your complete index and start from scratch if there are problems caused by this:
https://docs.photoprism.app/getting-started/updates/#face-recognition

**Reply by RobPattonDotCom** [reply]
No the re-deploy was to move it to a different database type, and I figured I had already spent 5 days, whats 4 more to do it clean. and I hadnt used docker compose to setup, just did a docker run...  No biggie. Really enjoy and am impressed with the software, when it isnt giving me errors.

**Reply by lastzero** [reply]
Ok, but our docs clearly recommend MariaDB, so that's not our fault then?
